src/fidesops/schemas/masking/masking_secrets.py

```diff
@@ -0,0 +1,39 @@
+from abc import ABC
+from dataclasses import dataclass
+from enum import Enum
+from typing import TypeVar, Generic, Callable, Optional
+
+from fidesops.schemas.base_class import BaseSchema
+
+
+T = TypeVar("T")
+
+
+class SecretType(Enum):
+    """Enum that holds all possible types of secrets across all masking strategies"""
+
+    key = "key"
+    salt = "salt"
+    # the below types are used by the AES algorithm, when it calls HMAC to generate the nonce
+    key_hmac = "key_hmac"
+    salt_hmac = "salt_hmac"
+
+
+@dataclass(unsafe_hash=True)
+class MaskingSecretMeta(Generic[T]):
```

Comment:
Why do we need `Generic[T]` here instead of the default?

Reply:
By using a `Generic[T]`, we can pass in the expected data type that `generate_secret` returns, when we're instantiating `MaskingSecretMeta`. Is this what you mean?